### ISSUE TITLE
Navigation to deeper levels

### ISSUE THREAD

**Issue by fulv** [plone/plone.restapi#505]
Question:
Is there any way to get a navigation menu structure that goes deeper than just the top level in a single call?    I'm thinking here of something that would allow a client to build a nav menu similar to what webcoutourier.dropdownmenu provides.  In order to build the dropdowns, you need to recurse to deeper levels, which ideally should be configurable.  

Or is the only option right now to get `@navigation` and then make another call for each of the items returned in the first call?

**Comment by tisto** [comment]
Not right now. This would be possible but I am not sure it is really needed. Retrieving the full content tree for the navigation portlet or a dropdown menu is possible but it does not really scale. I implemented a PoC for AngularJS long time ago and the calls that I used were for single levels (lazy loading the next level when the users clicks on a navigation item) and retrieving the full path to the portal root (for first time accessing a deeply nested page and getting an "expanded" navigation tree). 

Bottom line is I am not opposed to adding such a feature. Maybe I am just missing the use case here though...

**Comment by fulv** [comment]
My use case is there are two sites that have to exist side-by-side, one
Plone, one Wordpress.  The WP site needs to have the same nav menu as the
Plone site, and the Plone site has a menu rendered by
webcoutourier.dropdownmenu.  So this is not really a client-side
application use case, but the WP developer feels comfortable making a REST
API call to render the menu, and if plone.restapi allowed retrieving one or
two levels deeper than just the root navigation we'd be set.
I'll create a separate package as a POC that depends on
webcoutourier.dropdownmenu and plone.restapi.

Another thought I had:  is there some endpoint to get the sitemap?  I
couldn't find one.  The tricky thing here is that the content tree that is
needed must also take the exclude_from_navigation setting of each node into
account.

**Comment by fulv** [comment]
Resolved by #523.
Also, for the record, I did implement my POC in https://github.com/collective/collective.restapi.navigationtree.
